crates/matrix-sdk-base/src/sliding_sync/mod.rs

```diff
@@ -163,16 +161,11 @@ impl BaseClient {
         let store = self.store.clone();
         let mut ambiguity_cache = AmbiguityCache::new(store.inner.clone());
 
-        let account_data = &extensions.account_data;
-        let global_account_data_events = if !account_data.is_empty() {
-            self.handle_account_data(&account_data.global, &mut changes).await
-        } else {
-            Vec::new()
-        };
+        let account_data = AccountDataProcessor::process(&extensions.account_data.global);
 
         let mut new_rooms = RoomUpdates::default();
         let mut notifications = Default::default();
-        let mut rooms_account_data = account_data.rooms.clone();
+        let mut rooms_account_data = extensions.account_data.rooms.clone();
```

Comment:
Presumably this will go away too eventually and only the Processor/AccountData(?) will be used?

Reply:
I guess so, either we add room account data to the processor too or we create a different processor for this data, right?

Reply:
Yep, it would be another processor indeed.